internal/compiler/passes/ensure_window.rs

```diff
@@ -99,6 +99,22 @@ pub fn ensure_window(
         }
     });
 
+    // Fix up any ElementReferences, for example for builtin member function calls, to not refer to the WindowItem,
+    // as we swapped out the base_type.
+    let fixup_element_reference = |expr: &mut Expression| match expr {
+        Expression::ElementReference(elr)
+            if elr.upgrade().map_or(false, |elemrc| Rc::ptr_eq(&elemrc, &win_elem)) =>
+        {
+            *expr = Expression::ElementReference(Rc::downgrade(&new_root))
+        }
+        _ => {}
+    };
+
+    crate::object_tree::visit_all_expressions(component, |expr, _| {
```

Comment:
This is, in retrospect, a bad choice as it is very slow :(

Reply:
how slow is it?
We do that in many other passes, in particular, crate::object_tree::visit_all_named_references will do something similar.

Reply:
Maybe it was too late in the day, but it felt that the compiler was slower in the branch than in master when I ran this to see the impact on the generated code for the de-inlining.

Either way, I realize that this isn't the correct change either: Only element references that are in the arguments of a function call for a `ItemMemberFunction` need to be changed, as _those_ indirectly rely on the `base_type` of the referenced element, which is what's changing. There may be other element references that should correctly remain as-is perhaps. So the swap should only be done if `base_type` is `ElementType::Native`.